### PR TITLE
feat: dcmaw-8622 add developer menu

### PR DESCRIPTION
**Pull request by jdubey**
# DCMAW-8622: iOS | Authenticated Requests | Create developer menu and make it accessible from Home page

Add an empty Developer Menu modal view, accessible via the Tokens View after logging in.  This will only be accessible in DEBUG builds.

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
~~- [ ] Created a `draft` pull request if it is not yet ready for review~~

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
    - [x] Checked dynamic type sizes are applied
    - [x] Checked VoiceOver can navigate your new code
    - [x] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: Sources/Screens/DeveloperMenu/DeveloperMenuViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import GDSCommon
+import UIKit
+
+struct DeveloperMenuViewModel: BaseViewModel {
+    var rightBarButtonTitle: GDSCommon.GDSLocalisedString? = "Close"
```
Comment (DylanRJ) [inline Sources/Screens/DeveloperMenu/DeveloperMenuViewModel.swift:5]: Might want to remove this namespace to keep consistent with other view models

File: Sources/Tabs/HomeCoordinator.swift [hunk 2]
```diff
@@ -8,13 +8,24 @@ final class HomeCoordinator: NSObject,
...
     var root = UINavigationController()
     private var accessToken: String?
-    private let baseVc = TokensViewController()
+    private var baseVc: TokensViewController?
 
     func start() {
-        root.setViewControllers([baseVc], animated: true)
+        let tokensViewController = TokensViewController(TokensViewModel {
+            self.showDeveloperMenu()
+        })
```
Comment (DylanRJ) [inline Sources/Tabs/HomeCoordinator.swift:16]: Dya mind if we initialise the view model on the line above the view controller and pass in? Syntax here is a bit funky to read

File: Tests/UnitTests/Screens/TokensViewControllerTests.swift [hunk 3]
```diff
@@ -5,17 +5,22 @@ import XCTest
...
+    var didTapDeveloperMenu: Bool!
     
     override func setUp() {
         super.setUp()
         
-        sut = TokensViewController()
+        didTapDeveloperMenu = false
+        sut = TokensViewController(TokensViewModel {
+            self.didTapDeveloperMenu = true
+        })
```
Comment (DylanRJ) [inline Tests/UnitTests/Screens/TokensViewControllerTests.swift:16]: Same to the other comment about initialising the view model and passing in above if you don't mind

File: Sources/Screens/Tokens/TokensViewController.swift [hunk 4]
```diff
@@ -13,7 +14,8 @@ class TokensViewController: UIViewController {
         }
     }
     
-    init() {
```
Comment (DylanRJ) [inline Sources/Screens/Tokens/TokensViewController.swift:16]: Might be nice to have the parameter name just for clarity & consistence if ya don't mind

File: Sources/Screens/Tokens/TokensViewController.swift [hunk 5]
```diff
@@ -13,7 +14,8 @@ class TokensViewController: UIViewController {
...
+    init(_ viewModel: TokensViewModel) {
```
Comment (DylanRJ) [inline Sources/Screens/Tokens/TokensViewController.swift:17]: Might be nice to have the parameter name just for clarity & consistence if ya don't mind

**Review by DylanRJ — APPROVED** [review]

**Review by mcgarj — APPROVED** [review]

**Pull request merged**